File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.cpp
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> nums)
{
    int answer = 0;
    int max = nums.size() / 2;
    sort(nums.begin(), nums.end());
    nums.erase(unique(nums.begin(), nums.end()), nums.end());
    int ponketmon = nums.size();
    if (ponketmon >= max)
        answer += max;
    else
        answer += ponketmon;
    return answer;
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.py
def solution(nums):
    answer = 0
    dic = {}
    num_poketmon = len(nums) // 2
    for num in nums :
        if num in dic :
            dic[num] += 1
        else :
            dic[num] = 1
    if num_poketmon <= len(dic) :
        return num_poketmon
    else :
        return len(dic)
    return answer